internal/controller/provisioner/objects.go

```diff
@@ -43,6 +43,11 @@ const (
 	defaultInitialDelaySeconds = int32(3)
 )
 
+type PortInfo struct {
+	Port     int32
+	Protocol corev1.Protocol
+}
+
```

Comment:
Just curious, why did you make a struct for this?
From the way this is being used, it feels like you can reference Protocol on its own.

For example, on line `144`, we can define `ports := make(map[int32]corev1.Protocol)`
Line `155` then becomes `ports[int32(listener.Port)] = protocol`

Then the loop on line 473 becomes this

```go
for port, protocol := range ports {
  servicePort := corev1.ServicePort{
	  Name:       fmt.Sprintf("port-%d", port),
	  Port:       port,
	  TargetPort: intstr.FromInt32(port),
	  Protocol:   protocol,
```

Would love to know what you think though. Do please tell me if I'm overlooking anything.